src/Trinity.Core/Utilities/MISC/AssemblyPath.cs

```diff
@@ -58,9 +58,8 @@ internal static string MyAssemblyPath
                             if (asm == trinity_asm) continue;
                             if (asm.IsDynamic) continue;
                             if (type == typeof(System.RuntimeMethodHandle)) continue;
-                            if (type.FullName == "System.Linq.Enumerable+WhereSelectEnumerableIterator`2") continue;
-                            if (type.FullName == "System.Linq.Enumerable+WhereEnumerableIterator`1") continue;
-                            if (type.FullName == "System.Linq.Enumerable") continue;
+                            if (asm == typeof(object).Assembly) continue;
+                            if (asm == typeof(System.Linq.Enumerable).Assembly) continue;
```

Comment:
Yes, when the test fails, the `type` was `System.RuntimeMethodHandle` and its containing assembly was `mscorlib`, which was where `object` is defined.